### PR TITLE
DNN-17335: call to normal schedul update process when schedule enable status changed.

### PR DESCRIPTION
**Pull request by zyhfish**

### CODE REVIEW

File: src/Modules/Settings/Dnn.PersonaBar.TaskScheduler/Services/TaskSchedulerController.cs [hunk 1]
```diff
@@ -362,12 +362,13 @@ public HttpResponseMessage UpdateScheduleItem(ScheduleDto scheduleDto)
...
 
-                if ((updatedItem.ScheduleStartDate != existingItem.ScheduleStartDate) ||
-                    (updatedItem.Enabled) ||
-                    (updatedItem.TimeLapse != existingItem.TimeLapse) ||
-                    (updatedItem.RetryTimeLapse != existingItem.RetryTimeLapse) ||
-                    (updatedItem.RetryTimeLapseMeasurement != existingItem.RetryTimeLapseMeasurement) ||
-                    (updatedItem.TimeLapseMeasurement != existingItem.TimeLapseMeasurement))
+                if (updatedItem.ScheduleStartDate != existingItem.ScheduleStartDate ||
+                    updatedItem.Enabled ||
+                    updatedItem.Enabled != existingItem.Enabled ||
```
Comment (tpluscode) [inline src/Modules/Settings/Dnn.PersonaBar.TaskScheduler/Services/TaskSchedulerController.cs:367]: Is it relevant to check whether the change is from disabled to enabled or the other way round?

File: src/Modules/Settings/Dnn.PersonaBar.TaskScheduler/Services/TaskSchedulerController.cs [hunk 2]
```diff
@@ -362,12 +362,13 @@ public HttpResponseMessage UpdateScheduleItem(ScheduleDto scheduleDto)
...
+                    updatedItem.TimeLapse != existingItem.TimeLapse ||
+                    updatedItem.RetryTimeLapse != existingItem.RetryTimeLapse ||
+                    updatedItem.RetryTimeLapseMeasurement != existingItem.RetryTimeLapseMeasurement ||
+                    updatedItem.TimeLapseMeasurement != existingItem.TimeLapseMeasurement)
```
Comment (tpluscode) [inline src/Modules/Settings/Dnn.PersonaBar.TaskScheduler/Services/TaskSchedulerController.cs:371]: By the way, my head spins from all those statements. Maybe the condition can be simplified?

File: src/Modules/Settings/Dnn.PersonaBar.TaskScheduler/Services/TaskSchedulerController.cs (hunk 1 above)
Comment (zyhfish) [reply src/Modules/Settings/Dnn.PersonaBar.TaskScheduler/Services/TaskSchedulerController.cs:367]: change from disabled to enabled should always do the normal update.

File: src/Modules/Settings/Dnn.PersonaBar.TaskScheduler/Services/TaskSchedulerController.cs (hunk 2 above)
Comment (zyhfish) [reply src/Modules/Settings/Dnn.PersonaBar.TaskScheduler/Services/TaskSchedulerController.cs:371]: hmm, looks like need to check whether these properties has been changed one by one.

Comment (tpluscode) [reply src/Modules/Settings/Dnn.PersonaBar.TaskScheduler/Services/TaskSchedulerController.cs:371]: You say that we may be missing doing the wrong update in some cases?

Comment (zyhfish) [reply src/Modules/Settings/Dnn.PersonaBar.TaskScheduler/Services/TaskSchedulerController.cs:371]: I can't find out a better way to check all these conditions. :(

**Comment by tpluscode** [comment]
fixes #556

**Pull request merged**